### PR TITLE
Paella player 7

### PR DESCRIPTION
**Pull request by miesgre**
This PR adds a new module to opencast (`opencast-engage-paella-player-7`). This module adds paella7 to opencast.

`opencast-engage-paella-player-7` and `opencast-engage-paella-player` bundles can be installed together without problems, as the new bundle uses a new URL `/paella7`.

### Your pull request should…

* [x] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

**Comment by mliradelc** [comment]
It seems that there are some trailing spaces in the pom.xml file, can you fix that?

**Comment by miesgre** [comment]
> It seems that there are some trailing spaces in the pom.xml file, can you fix that?

Done!

**Comment by mliradelc** [comment]
Thanks Miguel, but now there are errors in these files:

```
Checking pom.xml files for tabs or trailing spaces…
+ echo 'Trailing spaces found in config files!'
etc/ui-config/mh_default_org/paella7/config.json:21:    
+ ret=1
etc/ui-config/mh_default_org/paella7/config.json:46:        
+ echo Checking that all modules include a build number…
etc/ui-config/mh_default_org/paella7/config.json:79:        
etc/ui-config/mh_default_org/paella7/config.json:85:        
etc/ui-config/mh_default_org/paella7/config.json:92:        
Trailing spaces found in config files!
```

File: .gitignore
Comment (ziegenberg) [inline .gitignore]:
```suggestion
# Ignore python cache folder
```
I guess this should reade "ignore".

File: docs/guides/admin/docs/modules/paella.player7/configuration.md
Comment (ziegenberg) [inline docs/guides/admin/docs/modules/paella.player7/configuration.md]:
```suggestion
- Easier styling and accessibility support.
```
Typo

File: docs/guides/admin/docs/modules/paella.player7/configuration.md
Comment (ziegenberg) [inline docs/guides/admin/docs/modules/paella.player7/configuration.md]:
```suggestion
- Reduce the number of dependencies.
```
Typo

File: docs/guides/admin/docs/modules/paella.player7/configuration.md
Comment (ziegenberg) [inline docs/guides/admin/docs/modules/paella.player7/configuration.md]:
```suggestion
Have a look at the paella 7 [repository](https://github.com/polimediaupv/paella-core)
```
Style

File: docs/guides/admin/docs/modules/paella.player7/configuration.md
Comment (ziegenberg) [inline docs/guides/admin/docs/modules/paella.player7/configuration.md]:
```suggestion
To activate the player set for each tenant the property `prop.player` in the file `.../etc/org.opencastproject.organization-<tenant>.cfg`.
```
Style

File: docs/guides/developer/docs/modules/paella-player-7/index.md
Comment (ziegenberg) [inline docs/guides/developer/docs/modules/paella-player-7/index.md]: Can you name the minimum version required?

File: docs/guides/developer/docs/modules/paella-player-7/index.md
Comment (ziegenberg) [inline docs/guides/developer/docs/modules/paella-player-7/index.md]:
```suggestion
By default, Paella Player will point to `http://localhost:8080` to redirect all the requests to Opencast, so if you compile and run Opencast on your local machine. To do it, you simply have to execute the following npm command:
```

File: docs/guides/developer/docs/modules/paella-player-7/index.md
Comment (ziegenberg) [inline docs/guides/developer/docs/modules/paella-player-7/index.md]:
```suggestion
But you also can debug and develop Paella Player without having an Opencast installation on your local machine. To do it, simply specify the `server` environment variable when you launch the npm command:
```

File: docs/guides/developer/docs/modules/paella-player-7/index.md
Comment (ziegenberg) [inline docs/guides/developer/docs/modules/paella-player-7/index.md]:
```suggestion
Or you can use this command to execute all the previous commands together:
```

File: modules/engage-paella-player-7/.babelrc
Comment (ziegenberg) [inline modules/engage-paella-player-7/.babelrc]: Do you want to add a comment why this empty file is included?

**Review by ziegenberg — COMMENTED** [review]
I've commented some typos and style suggestions.

File: modules/engage-paella-player-7/.babelrc
Comment (miesgre) [reply modules/engage-paella-player-7/.babelrc]: This file is included because `eslint`. Try to remove the file and run `npm run eslint`
```
0:0  error  Parsing error: No Babel config file detected for /opencast/modules/engage-paella-player-7/src/index.js. Either disable config file checking with requireConfigFile: false, or configure Babel so that it can find the config files
```

I can add the `requireConfigFile: false` to the `.eslintrc.js` file, so this file is not needed. Or add `babel: {}` to the `package.json` file.

File: docs/guides/developer/docs/modules/paella-player-7/index.md
Comment (miesgre) [reply docs/guides/developer/docs/modules/paella-player-7/index.md]: @ferserc1, Can you answer this question?

Comment (ferserc1) [reply docs/guides/developer/docs/modules/paella-player-7/index.md]: I have modified the documentation to specify Node.js version 14 as the minimum version required.

File: modules/engage-paella-player-7/.babelrc
Comment (miesgre) [reply modules/engage-paella-player-7/.babelrc]: I removed the .babelrc file

File: etc/ui-config/mh_default_org/paella7/present-mode-1.svg
Comment (ziegenberg) [inline etc/ui-config/mh_default_org/paella7/present-mode-1.svg]:
```suggestion
    <path d="m36 34h-36v-34h36zm-16-12 5.3 8.5c0.73 1.2-0.99 2.2-1.7 1.1l-5-8.1-5.2 8.1c-0.73 1.1-2.6 0.059-1.8-1.1l5.4-8.5h-12v-17h-2v-2h30v2h-2v17zm-13-17h22v15h-22zm8.5 5.1-6.3 7.3-0.62-0.52 6.9-7.9 4.5 4.5 6.2-6 0.58 0.57-6.8 6.6z"/>
```

I smoothed the corners of the stand.

File: etc/ui-config/mh_default_org/paella7/present-mode-3.svg
Comment (ziegenberg) [inline etc/ui-config/mh_default_org/paella7/present-mode-3.svg]:
```suggestion
    <path d="m36 34h-36v-34h36zm-16-12 5.3 8.5c0.73 1.2-0.99 2.2-1.7 1.1l-5-8.1-5.2 8.1c-0.73 1.1-2.6 0.059-1.8-1.1l5.4-8.5h-12v-17h-2v-2h30v2h-2v17zm-13-17h22v15h-22zm8.5 5.1-6.3 7.3-0.62-0.52 6.9-7.9 4.5 4.5 6.2-6 0.58 0.57-6.8 6.6zm58.5 23.9h-36.5v-34h36.5zm-10.5-16.9h-18c-0.48 0-0.86 0.38-0.86 0.86v1.3c0 0.48 0.38 0.86 0.86 0.86h5v11h8v-11h5c0.47 0 0.86-0.38 0.86-0.86v-1.3c0-0.48-0.38-0.86-0.86-0.86zm-2-1c-1e-3 -0.016-0.44-4.7-1.1-6-0.62-1.3-3.7-1.3-3.7-1.3s-0.71 3.3-0.9 3.9c-0.19 0.63-0.23 0.78-0.23 0.78s-0.48-3.2-0.56-3.5c-0.062-0.24-0.47-0.26-0.56-0.26s-0.5 0.028-0.56 0.27c-0.083 0.32-0.43 3.4-0.43 3.4s-0.1-0.075-0.29-0.7c-0.19-0.63-0.97-3.9-0.97-3.9s-3.1 0.045-3.7 1.3c-0.62 1.3-1.1 6-1.1 6zm-5.9-7.3c0.061 0.26-0.77 0.82-0.77 0.82l-0.66-0.016s-0.88-0.55-0.82-0.81c0.056-0.24 0.81-0.21 1.1-0.22 0.32 3e-3 1.1-0.023 1.1 0.22zm-1.1-7.6c1.7 0 3.1 1.5 3.1 3.4s-1.4 3.4-3.1 3.4-3.1-1.5-3.1-3.4 1.4-3.4 3.1-3.4z"/>
```

I smoothed the corners of the stand.

File: etc/ui-config/mh_default_org/paella7/present-mode-4.svg
Comment (ziegenberg) [inline etc/ui-config/mh_default_org/paella7/present-mode-4.svg]:
```suggestion
    <path d="m22.6 34.6h-19.9v-14.7h19.9zm-9.1-5.2 2.3 3.68c0.316 0.52-0.429 0.954-0.737 0.477l-2.17-3.51-2.25 3.51c-0.316 0.477-1.13 0.0256-0.78-0.477l2.34-3.68h-5.2v-7.37h-0.867v-0.867h13v0.867h-0.867v7.37zm-5.64-7.37h9.54v6.5h-9.54zm3.68 2.21-2.73 3.16-0.269-0.225 2.99-3.42 1.95 1.95 2.69-2.6 0.251 0.247-2.95 2.86z" stroke-width=".434"/>
    <path d="m42.3 18.9h-9.55v2.46h-8.45v-2.46h-9.52v-17.9h27.5zm-8.32-7.71h-10.9c-0.287 0-0.522 0.23-0.522 0.515v0.766c0 0.284 0.235 0.515 0.522 0.515h3.01v6.58h4.85v-6.58h3.01c0.287 0 0.522-0.23 0.522-0.515v-0.766c0-0.284-0.235-0.515-0.522-0.515zm-1.2-0.598c-0.0012-0.0096-0.267-2.8-0.64-3.57-0.375-0.764-2.24-0.791-2.24-0.791s-0.429 1.98-0.543 2.36c-0.116 0.377-0.139 0.464-0.139 0.464s-0.291-1.91-0.342-2.11c-0.0377-0.146-0.288-0.156-0.341-0.156-0.0533 0-0.303 0.0168-0.342 0.162-0.0503 0.194-0.259 2.06-0.259 2.06s-0.0634-0.0449-0.178-0.422c-0.115-0.376-0.586-2.36-0.586-2.36s-1.86 0.0269-2.24 0.791c-0.374 0.765-0.64 3.57-0.64 3.57zm-3.56-4.36c0.0365 0.156-0.466 0.493-0.466 0.493l-0.402-0.00958s-0.535-0.327-0.499-0.483c0.0341-0.144 0.493-0.128 0.683-0.13 0.192 0.0018 0.649-0.0138 0.683 0.13zm-0.683-4.54c1.04 0 1.88 0.901 1.88 2.01s-0.843 2.01-1.88 2.01-1.88-0.9-1.88-2.01 0.843-2.01 1.88-2.01z" stroke-width=".598"/>
    <path d="m54.4 34.6h-19.9v-14.7h19.9v14.7zm-6.03-7.37h-7.88c-0.208 0-0.378 0.167-0.378 0.373v0.555c0 0.206 0.17 0.373 0.378 0.373h2.18v4.77h3.52v-4.77h2.18c0.208 0 0.378-0.167 0.378-0.373v-0.555c0-0.206-0.17-0.373-0.378-0.373zm-0.869-0.434c-8.68e-4 -0.0069-0.193-2.03-0.464-2.58-0.271-0.554-1.62-0.573-1.62-0.573s-0.311 1.43-0.394 1.71c-0.0837 0.273-0.101 0.336-0.101 0.336s-0.211-1.39-0.248-1.53c-0.0273-0.106-0.209-0.113-0.247-0.113-0.0386 0-0.22 0.0121-0.248 0.117-0.0364 0.141-0.188 1.49-0.188 1.49s-0.046-0.0325-0.129-0.306c-0.0832-0.273-0.424-1.71-0.424-1.71s-1.35 0.0195-1.62 0.573c-0.271 0.554-0.464 2.58-0.464 2.58zm-2.58-3.16c0.0264 0.113-0.337 0.357-0.337 0.357l-0.291-0.0069s-0.388-0.237-0.361-0.35c0.0247-0.104 0.357-0.0928 0.495-0.0941 0.139 0.0013 0.47-0.01 0.495 0.0941zm-0.495-3.29c0.752 0 1.36 0.652 1.36 1.46s-0.61 1.46-1.36 1.46-1.36-0.652-1.36-1.46 0.61-1.46 1.36-1.46z" stroke-width=".434"/>
```

I smoothed the corners of the stand. And for the icon in the lower-left, I removed the two white notches. I guess they were no intended.

**Review by ziegenberg — CHANGES_REQUESTED** [review]
I suggest some changes for the SVGs. Please have a look if you like them.

File: etc/ui-config/mh_default_org/paella7/present-mode-1.svg
Comment (miesgre) [reply etc/ui-config/mh_default_org/paella7/present-mode-1.svg]: Hi @ziegenberg, I'm not and expert with github revisions. There is an easy way to download your changes and apply in my local repository to view your changes?

Comment (ziegenberg) [reply etc/ui-config/mh_default_org/paella7/present-mode-1.svg]: You can find these suggested changes here: https://github.com/ziegenberg/opencast/commit/e86ec143a95e2b7e60c3212917286195e5fd5e3e

**Review by mliradelc — DISMISSED** [review]
When enter to the web console and entering the command `paella.version` I've got that the version is 6.4.4 instead of 7.

```
paella.version
'6.4.4 - build: f10e17e2b2'
```

**Comment by miesgre** [comment]
> When enter to the web console and entering the command `paella.version` I've got that the version is 6.4.4 instead of 7.
> 
> ```
> paella.version
> '6.4.4 - build: f10e17e2b2'
> ```

`paella.version` does not exists in paella 7
Which URL are you using? paella 7 bundle does not replace the paella URL. It adds a new URL `/paella7/ui/watch.html`

**Comment by mliradelc** [comment]
> > When enter to the web console and entering the command `paella.version` I've got that the version is 6.4.4 instead of 7.
> 
> > 
> 
> > ```
> 
> > paella.version
> 
> > '6.4.4 - build: f10e17e2b2'
> 
> > ```
> 
> 
> 
> `paella.version` does not exists in paella 7
> 
> Which URL are you using? paella 7 bundle does not replace the paella URL. It adds a new URL `/paella7/ui/watch.html`
> 
> 

Hi Miguel, 

Simply add a video to opencast, then open it with the engage. When you play the video, open the web console.

**Comment by mliradelc** [comment]
Because it's still in beta, I  added the `Beta` to the title. 

There is some problem in the paella 7 build, I think these are the problems:
```
[INFO] > eslint --config .eslintrc.js src --resolve-plugins-relative-to .
[INFO] 
[INFO] 
[INFO] /home/runner/work/opencast/opencast/modules/engage-paella-player-7/src/index.js
[INFO]   141:47  error  'evt' is defined but never used               no-unused-vars
[INFO]   142:45  error  Strings must use singlequote                  quotes
[INFO]   142:78  error  Strings must use singlequote                  quotes
[INFO]   151:1   error  Expected indentation of 6 spaces but found 8  indent
[INFO]   154:1   error  Expected indentation of 6 spaces but found 8  indent
[INFO]   157:1   error  Expected indentation of 6 spaces but found 8  indent
[INFO] 
[INFO] ✖ 6 problems (6 errors, 0 warnings)
[INFO]   5 errors and 0 warnings potentially fixable with the `--fix` option.
```

**Comment by miesgre** [comment]
> > > When enter to the web console and entering the command `paella.version` I've got that the version is 6.4.4 instead of 7.
> > 
> > 
> > > 
> > 
> > 
> > > 
> > 
> > 
> > > paella.version
> > 
> > 
> > > '6.4.4 - build: [f10e17e](https://github.com/opencast/opencast/commit/f10e17e2b2ec3656e56a69f4b006c87b0e5ec0fd)'
> > 
> > 
> > > 
> > 
> > 
> > `paella.version` does not exists in paella 7
> > Which URL are you using? paella 7 bundle does not replace the paella URL. It adds a new URL `/paella7/ui/watch.html`
> 
> Hi Miguel,
> 
> Simply add a video to opencast, then open it with the engage. When you play the video, open the web console.

@mliradelc, I don't understand what you are saying.

If you add a video to opencast and open it with the engage (without configure anything else), the player used is paella player 6.4.4 (the default player). To use paella 7 as the default player you need to change the `prop.player` in the file `.../etc/org.opencastproject.organization-<tenant>.cfg`.

    prop.player=/paella7/ui/watch.html?id=#{id}

Also, you can change the URL manually to point the paella player 7. 

    Ex: http://localhost:8080/paella7/ui/watch.html?id=19fa470b-0248-4bb2-a118-635807af5d39

When I open the video, my console says:
```
paella-core - Debug: New paella player instance
paella-core.js:2 paella-core - Debug: Loading paella player
paella-core.js:2 paella-core - Debug: Using default configuration loading function.
paella-core.js:2 paella-core - Info: Paella player load done
```

**Comment by mliradelc** [comment]
> > > > When enter to the web console and entering the command `paella.version` I've got that the version is 6.4.4 instead of 7.
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > paella.version
> > > 
> > > 
> > > > '6.4.4 - build: [f10e17e](https://github.com/opencast/opencast/commit/f10e17e2b2ec3656e56a69f4b006c87b0e5ec0fd)'
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > `paella.version` does not exists in paella 7
> > > Which URL are you using? paella 7 bundle does not replace the paella URL. It adds a new URL `/paella7/ui/watch.html`
> > 
> > 
> > Hi Miguel,
> > Simply add a video to opencast, then open it with the engage. When you play the video, open the web console.
> 
> @mliradelc, I don't understand what you are saying.
> 
> If you add a video to opencast and open it with the engage (without configure anything else), the player used is paella player 6.4.4 (the default player). To use paella 7 as the default player you need to change the `prop.player` in the file `.../etc/org.opencastproject.organization-<tenant>.cfg`.
> 
> ```
> prop.player=/paella7/ui/watch.html?id=#{id}
> ```
> 
> Also, you can change the URL manually to point the paella player 7.
> 
> ```
> Ex: http://localhost:8080/paella7/ui/watch.html?id=19fa470b-0248-4bb2-a118-635807af5d39
> ```
> 
> When I open the video, my console says:
> 
> ```
> paella-core - Debug: New paella player instance
> paella-core.js:2 paella-core - Debug: Loading paella player
> paella-core.js:2 paella-core - Debug: Using default configuration loading function.
> paella-core.js:2 paella-core - Info: Paella player load done
> ```

You are right! I assumed Paella player 7 as default (Because we are in the develop branch), but after thinking about,left Paella 7 Beta as an experimental alternative makes more sense.

**Comment by mliradelc** [comment]
The Java build has a problem,

```
Warning:  Error resolving artifact org.opencastproject:opencast-engage-paella-player-7:jar:12-SNAPSHOT: [Could not find artifact org.opencastproject:opencast-engage-paella-player-7:jar:12-SNAPSHOT in central (https://repo.maven.apache.org/maven2/), Could not find artifact org.opencastproject:opencast-engage-paella-player-7:jar:12-SNAPSHOT in mvn.opencast.org (https://mvn.opencast.org/)]
java.io.IOException: Error resolving artifact org.opencastproject:opencast-engage-paella-player-7:jar:12-SNAPSHOT: [Could not find artifact org.opencastproject:opencast-engage-paella-player-7:jar:12-SNAPSHOT in central (https://repo.maven.apache.org/maven2/), Could not find artifact org.opencastproject:opencast-engage-paella-player-7:jar:12-SNAPSHOT in mvn.opencast.org (https://mvn.opencast.org/)]
```

This issue was caused from the release cut last week, can you update the maven repo?  🙂

**Comment by miesgre** [comment]
> The Java build has a problem,
> 
> ```
> Warning:  Error resolving artifact org.opencastproject:opencast-engage-paella-player-7:jar:12-SNAPSHOT: [Could not find artifact org.opencastproject:opencast-engage-paella-player-7:jar:12-SNAPSHOT in central (https://repo.maven.apache.org/maven2/), Could not find artifact org.opencastproject:opencast-engage-paella-player-7:jar:12-SNAPSHOT in mvn.opencast.org (https://mvn.opencast.org/)]
> java.io.IOException: Error resolving artifact org.opencastproject:opencast-engage-paella-player-7:jar:12-SNAPSHOT: [Could not find artifact org.opencastproject:opencast-engage-paella-player-7:jar:12-SNAPSHOT in central (https://repo.maven.apache.org/maven2/), Could not find artifact org.opencastproject:opencast-engage-paella-player-7:jar:12-SNAPSHOT in mvn.opencast.org (https://mvn.opencast.org/)]
> ```
> 
> This issue was caused from the release cut last week, can you update the maven repo? 🙂

@mliradelc, Done! You can continue reviewing

**Comment by mliradelc** [comment]
Hi @miesgre, @ferserc1 and all the Polimedia team.

I want to thank you about your work on this new version of Paella 7. We have a discussion on the technical meeting over your PR because it is a Beta version, and we reach that the best is to leave this PR as a draft until it becomes to release state.

As a draft, it can be seeing for the users and anyone that want to test it can do it simply checking out the branch, but maybe there another options that you also want to try like **create a standalone Jar** to be installed later in opencast (Like the old Wowza plugin and the Opencast Annotation Tool), that would be very handy to try it on distributed installations.

File: docs/guides/admin/docs/modules/paella.player7/configuration.md
Comment (lkiesow) [inline docs/guides/admin/docs/modules/paella.player7/configuration.md:40]: Can you add the new option to the configuration file at:

https://github.com/opencast/opencast/blob/59188fb8b47055b6967577f04c40a969486a2cf4/etc/org.opencastproject.organization-mh_default_org.cfg#L178-L185

Comment (miesgre) [reply docs/guides/admin/docs/modules/paella.player7/configuration.md:40]: Done!

**Comment by mliradelc** [comment]
I'm testing right now under our dev cluster and I have a question, are the plugins from paella 6.x compatible with paella 7? I'm asking because I saw that the matomo plugin is not yet in the beta and I want to try to port it to Paella 7.

**Comment by mliradelc** [comment]
When Testing Paella 7 with safari, on a HLS stream, I've got this error:

`Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating 't._players[0].currentTimeSync')`



image



By the way, How I can enable the `hls.js` debug mode on Paella?

**Comment by mliradelc** [comment]
> When Testing Paella 7 with safari, on a HLS stream, I've got this error:
> 
> `Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating 't._players[0].currentTimeSync')`
> 
> image
> 
> By the way, How I can enable the `hls.js` debug mode on Paella?

I've found how to enable the debug mode, simply in the config.json file add `debug: true` in the HLS options section.

About HLS, we have still testing, but we found that some  HLS streams paella won't want to play, but any other player, even the built-in player safari, can play the stream without issues. Now I've found a tool from apple that checks if the HLS stream is within the standards. 

I put the streams with problems to analyze, and I've got that there are some issues  with the encoding/m3u8 file that they need to be fixed.

**Comment by miesgre** [comment]
> I'm testing right now under our dev cluster and I have a question, are the plugins from paella 6.x compatible with paella 7? I'm asking because I saw that the matomo plugin is not yet in the beta and I want to try to port it to Paella 7.

Hi Max.

Plugins are not compatible, but port matomo plugin should not be difficult.
As a example you can view the `opencast usertracking plugin` 
 - [usertracking in Paella 6](https://github.com/polimediaupv/opencast/blob/paella-player-7/modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.userTrackingSaverPlugIn/mh_usertracking_saver.js)
 - [usertracking in Paella 7](https://github.com/polimediaupv/opencast/blob/paella-player-7/modules/engage-paella-player-7/src/plugins/es.upv.paella.opencast.userTrackingDataPlugin.js)

As you can see, they are quite similar. We don't use the matomo plugin, so if you could port it, it would be wonderful.

**Comment by mliradelc** [comment]
@miesgre and @ferserc1 

After a few weeks of testing, we found a few bugs, should I published that bugs here or in a particular repository?

--------
I have a question regarding Spring security configuration for paella 7. Paella 6 have this spring security configuration:

```
<sec:intercept-url pattern="/paella/ui/auth.html" access="ROLE_USER" />
<sec:intercept-url pattern="/paella/ui/**" access="ROLE_ANONYMOUS" />
```

I've tried to recreate, but it doesn't work with Paella 7 (Doesn't open the login for the engage server, thus Paella can't find the event), but if add only `<sec:intercept-url pattern="/paella7/ui/**" access="ROLE_USER" />` Paella7 can work also for the rest of the users.

What would be the correct configuration for spring security?

**Comment by miesgre** [comment]
> @miesgre and @ferserc1
> 
> After a few weeks of testing, we found a few bugs, should I published that bugs here or in a particular repository?

If the bug is related to paella itself, please send the issue to paella repository. On the other hand, if it is related tho the opencast implementation publish the issue here.

**Comment by miesgre** [comment]
> I've tried to recreate, but it doesn't work with Paella 7 (Doesn't open the login for the engage server, thus Paella can't find the event), but if add only `<sec:intercept-url pattern="/paella7/ui/**" access="ROLE_USER" />` Paella7 can work also for the rest of the users.
> 
> What would be the correct configuration for spring security?

We forget to add paella7 the security section. I will add it.

**Comment by miesgre** [comment]
> The normal way is when you are not logged in the engage server have to ask you if the video is not set as anonymous, this part not happens in Paella 7

You were right. This part was not finished. I fixed it and it should work. Please test it.

**Comment by gregorydlogan** [comment]
Apologies, Github doesn't notify us that first-time committers have contributed so we didn't know that the tests weren't running.  If this happens again free free to tag a committer so that we can hit the magic button.

What's the status of this PR btw?  The header says Paella 7 is not ready for production, is that still true?

**Comment by miesgre** [comment]
> What's the status of this PR btw? The header says Paella 7 is not ready for production, is that still true?

There are some minor issues and we are working in some changes in the layout selector/configurator, but we think it can be used for massive testing.
We plan to activate paella 7 in our installations in a few days

**Comment by miesgre** [comment]
@lkiesow, @mliradelc Paella 7 in ready for review.

This PR does not add paella 7 as the default player. You need to change the URL manually to point the paella player 7 (`/paella7/ui/watch.html`) or configure as the default player.

**Comment by mliradelc** [comment]
Thanks @miesgre and @ferserc1 for the outstanding work 🎉.

Can you rebase the branch and squash the commits? it will be easier to review and to avoid any conflicts. 

By the way:

> Warning
Paella 7 is still in beta and this module should not be used in production.
It is intended to test the new features of paella 7.


Should be removed the warning from the description?

**Comment by miesgre** [comment]
> Can you rebase the branch and squash the commits? it will be easier to review and to avoid any conflicts.

Done! 
 
> By the way:
> 
> > Warning
> > Paella 7 is still in beta and this module should not be used in production.
> > It is intended to test the new features of paella 7.
> 
> Should be removed the warning from the description?

Yes. we have done some tests and everything seems to work correctly.
On our side, we have installed paella 7 in our installation.

File: docs/guides/admin/docs/modules/paella.player7/configuration.md
Comment (mliradelc) [inline docs/guides/admin/docs/modules/paella.player7/configuration.md]: Now that is ready, you can delete that part. :)

**Review by mliradelc — CHANGES_REQUESTED** [review]
I have only one thing that I need you to modify.

Also, It would be great if you can add Paella 7 to the [Opencast 12 release notes ](https://github.com/opencast/opencast/blob/develop/docs/guides/admin/docs/releasenotes.md)

File: docs/guides/admin/docs/modules/paella.player7/configuration.md
Comment (miesgre) [reply docs/guides/admin/docs/modules/paella.player7/configuration.md]: Done!

**Comment by miesgre** [comment]
@mliradelc I updated the PR.

**Review by mliradelc — CHANGES_REQUESTED** [review]
Thanks @miesgre !, I want to merge this PR after the Technical meeting today, can you do 2 
 little things:
 - Add a little txt about paella like the other features that are already here https://github.com/opencast/opencast/tree/develop/docs/guides/admin/docs/releasenotes
 - ~Squash the latest commits~ I forgot that I can do that while merging 😋  

Thanks for the work!

**Review by mliradelc — APPROVED** [review]

**Comment by mliradelc** [comment]
All good, I will do the merge after the tech meeting

**Comment by lkiesow** [comment]
@miesgre, this unfortunately breaks the build of Opencast when using a local npm since the commands you  defined in your modules `pom.xml` file do not actually exist:

https://github.com/opencast/opencast/blob/2247c0117426d4e8e36e31bc331ebd4f92f85430/modules/engage-paella-player-7/pom.xml#L46-L47

For more details, take a look at #3629. Can you take a look at that and fix the problem?

**Pull request merged**
